pkg/controller/notification/notification_controller.go

```diff
@@ -224,15 +239,28 @@ func (c *Controller) reconcile(obj interface{}) error {
 	}
 
 	name := accessor.GetName()
-	kind := runtimeObj.GetObjectKind().GroupVersionKind().String()
+
+	// If the cluster changed, add or delete, it should update the annotations of secrets
```

Comment:
=>
	// The notification controller should update the annotations of secrets managed by itself
         // whenever a cluster is added or deleted. This way, the controller will have a chance to override the secret.